Sources/MapboxCoreNavigation/NavigationRouter.swift

```diff
@@ -0,0 +1,337 @@
+@_implementationOnly import MapboxCommon_Private
+import MapboxDirections
+import MapboxNavigationNative
+
+/**
+ Provides alternative access to routing API.
+ 
+ Use this class instead `Directions` requests wrapper to request new routes or refresh an existing one. Depending on `RouterSource`, `NavigationRouter` will use online and/or onboard routing engines. This may be used when designing purely online or offline apps, or when you need to provide best possible service regardless of internet collection.
+ */
+public class NavigationRouter: NavigationProvider {
```

Comment:
NavigationRouter is unrelated to the Router protocol, so we need to find a different name. I’ve wanted to rename the Router protocol to something less confusing for a long time, but I think further overloading the word in the meantime will only increase confusion.